std/assembly/typedarray.ts

```diff
@@ -94,27 +94,27 @@ export class Int8Array extends ArrayBufferView {
     return REDUCE_RIGHT<Int8Array, i8, T>(this, fn, initialValue);
   }
 
-  map(fn: (value: i8, index: i32, self: Int8Array) => i8): Int8Array {
+  map(fn: (value: i8, index: i32, self: this) => i8): Int8Array {
```

Comment:
In AS, the `this` type is static, so here it is always `Int8Array` regardless of subtyping. As such, with this change, TS would infer something different than AS when the method is overloaded, which is probably not what you want.

Reply:
Ok, Maybe I should close this PR :(.